src/test/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,267 @@
+        if (ring != 1) {
+            resultGrid = upsideDownFlipInternalMatrix(this.squarelotron);
+        } else {
+            //complete Upside Down Flip.
+            resultGrid = upsideDownFlipCompleteMatrix(this.squarelotron);
+
+            // Restore values of the internal rings if there are.
+            for (int i = 1; i < size - 1; i++) {
+                for (int j = 1; j < size - 1; j++) {
+                    resultGrid[i][j] = this.squarelotron[i][j];
+                }
+            }
+        }
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * flipExternalRing.
+     *
+     * @param matrix to be flip.
+     * @return the matrix result.
+     */
+    private int[][] flipExternalRing(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
+        int tempJ = 0;
+        for (int i = 1; i < size; i++) {
+            resultGrid[i][tempJ] = matrix[tempJ][i];
+            resultGrid[tempJ][i] = matrix[i][tempJ];
+        }
+
+        int tempI = size - 1;
+        for (int j = 1; j < size - 1; j++) {
+            resultGrid[tempI][j] = matrix[j][tempI];
+            resultGrid[j][tempI] = matrix[tempI][j];
+        }
+        return resultGrid;
+    }
+
+    /**
+     * flipInternalRing.
+     *
+     * @param matrix to be flip.
+     * @return matrix result
+     */
+    private int[][] flipInternalRing(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
+        int tempJ = 1;
+        for (int i = 1; i < size - 1; i++) {
+            resultGrid[i][tempJ] = matrix[tempJ][i];
+            resultGrid[tempJ][i] = matrix[i][tempJ];
+        }
+
+        int tempI = size - 2;
+        for (int j = 1; j < size - 2; j++) {
+            resultGrid[tempI][j] = matrix[j][tempI];
+            resultGrid[j][tempI] = matrix[tempI][j];
+        }
+        return resultGrid;
+    }
+
+    /**
+     * mainDiagonalFlip.
+     *
+     * @param ring number.
+     * @return the Squarelotron.
+     */
+    Squarelotron mainDiagonalFlip(int ring) {
+        int[][] resultGrid = null;
+        if (this.size % 2 == 1 && ring == this.numberOfRings) {
+            return this; // returns the same squarelotron if the ring number is equal to the number of rings.
+        }
+        if (ring != 1) {
+            resultGrid = flipInternalRing(this.squarelotron);
+        } else {
+            resultGrid = flipExternalRing(this.squarelotron);
+        }
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * Rotates the matrix.
+     *
+     * @param numberOfTurns number.
+     */
+    void rotateRight(int numberOfTurns) {
+        final int zeroNumberOfTurns = 0;
+        final int maxNumberOfTurns = 4;
+        if (numberOfTurns < maxNumberOfTurns && numberOfTurns != 0) {
+            int[][] resultGrid = null;
+            Squarelotron squarelotron = new Squarelotron(this.squarelotron);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "squarelotron" which hides the field declared at line 9. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)